### PR TITLE
Add stronger typing to gradient accumulation scheduler callback

### PR DESCRIPTION
**Pull request by ananthsub**
## What does this PR do?
- Add types for gradient accumulation scheduler callback
- `scheduling` is a `Dict[int, int]`
- add types for the `on_epoch_start`
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

**Comment by awaelchli** [comment]
@ananthsub it is failing because there are circular imports

**Review by awaelchli — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

**Comment by ananthsub** [comment]
@williamFalcon can this be merged?
